### ISSUE TITLE
Tries to make enormous image when passed annotations instead of texts

### ISSUE THREAD

**Issue by alphapapa** [Phlya/adjustText#7]
Hi again, quick bug report: when I pass `adjust_text` a list of annotations made with `plot.annotate()`, it tries to make an enormous image, over 900,000 pixels wide.  When I switched to passing it a list of objects made with `plot.text()`, it worked.  I think this is probably a bug, but I'm not sure.  Thanks.

**Comment by Phlya** [comment]
Hi again, this is not a bug, it's currently not intended to work with annotation objects, but would be great to make this possible. I don't have time to work on this now, but would be happy to consider a PR for this.

**Comment by Phlya** [comment]
Should be quite similar, you just need to figure out how to get bbox of the text part of the annotation...

**Comment by alphapapa** [comment]
Ok, thanks.  I might take a look at this again in a few weeks.  :)

**Comment by Phlya** [comment]
Shouldn't be too complex I guess... You just need to get bbox from the text
of the annotation, not the whole annotation, I guess it should be possible
somehow.

2016-10-22 5:15 GMT+01:00 alphapapa <notifications@github.com>:

**Comment by yech1990** [comment]
any progress?

**Comment by Phlya** [comment]
No, sorry... any particular reason you need to use annotations?

**Comment by yech1990** [comment]
another case

before
![image](https://user-images.githubusercontent.com/5415510/46566963-7cee5080-c95b-11e8-8277-196a02f53047.png)

after
![image](https://user-images.githubusercontent.com/5415510/46566952-0a7d7080-c95b-11e8-8585-a5150d8dcfea.png)

manually adjusting the annotation label make it better.

In spite of the location of text label is adjusted, it will mess up either. Cause there is no boarder is numbers, when the digits are too close, it is difficult to distinguish.

eg

123 45 vs 12 345

**Comment by yech1990** [comment]
The main difference between `ax.text` and `ax.annotation` is that the text is for points only and annotation is suitable for higher dimension geometric elements.

**Comment by Phlya** [comment]
Have you tried just changing the bbox properties of text before adjusting
it? And I think arrows with an angle can be specified with arrowprops in
adjust_text?

Sorry, I'm travelling without a laptop, so can't try it myself.

**Comment by Conceptron** [comment]
Hi, is there any workaround for this? Maybe a snippet to get the text from annotations would help. Thanks
